compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -396,34 +398,53 @@ impl<'a, 'tcx> AbstractConstBuilder<'a, 'tcx> {
             | ExprKind::Field { .. }
             | ExprKind::ConstBlock { .. }
             | ExprKind::Adt(_) => self.error(
+                Some(node.span),
+                "unsupported operation in generic constant, this may be supported in the future",
+            )?,
+            ExprKind::Pointer { .. } =>
+            // dont know if this is correct
+            {
+                self.error(Some(node.span), "pointers are not allowed in generic constants")?
+            }
+            ExprKind::Continue { .. } | ExprKind::Break { .. } | ExprKind::Loop { .. } => self
+                .error(
                     Some(node.span),
-                    "unsupported operation in generic constant, this may be supported in the future",
+                    "loops and loop control flow are not supported in generic constants",
                 )?,
+            ExprKind::Box { .. } =>
+            // allocations not allowed in constants
```

Comment:
we can probably remove this comment now that there's an error message right next to it saying the same thing